<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>BiocAlphaMissense: Bioconductor interfaces to AlphaMissense pathogenicity findings • BiocAlphaMissense</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="BiocAlphaMissense: Bioconductor interfaces to AlphaMissense pathogenicity findings">
<meta property="og:description" content="BiocAlphaMissense">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BiocAlphaMissense</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/BiocAlphaMissense.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/vjcitn/BiocAlphaMissense/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="BiocAlphaMissense_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>BiocAlphaMissense: Bioconductor interfaces to AlphaMissense pathogenicity findings</h1>
                        <h4 data-toc-skip class="author">Vincent J. Carey, stvjc at channing.harvard.edu</h4>
            
            <h4 data-toc-skip class="date">September 22, 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/vjcitn/BiocAlphaMissense/blob/HEAD/vignettes/BiocAlphaMissense.Rmd"><code>vignettes/BiocAlphaMissense.Rmd</code></a></small>
      <div class="hidden name"><code>BiocAlphaMissense.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>AlphaMissense (<a href="https://www.science.org/doi/epdf/10.1126/science.adg7492" class="external-link">Science 2023</a>) is a machine learning procedure for inferring pathogenicity of single nucleotide substitutions in the human proteome. Text files are distributed by the authors, and these have been tabix-indexed for use in this package. Note that the pathogenicity data are produced under a license that forbids commercial use.</p>
<p>The package described here can currently be installed using</p>
<pre><code><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"vjcitn/BiocAlphaMissense"</span><span class="op">)</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="a-quick-look-at-single-nucleotide-data">A quick look at single-nucleotide data<a class="anchor" aria-label="anchor" href="#a-quick-look-at-single-nucleotide-data"></a>
</h2>
<p>The following code draws the first 10 records and places them in a data.frame. Note that 600 MB of bgzipped text and a tabix index will be retrieved and cached on first attempt.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/vjcitn/BiocAlphaMissense">BiocAlphaMissense</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Rsamtools" class="external-link">Rsamtools</a></span><span class="op">)</span></span>
<span><span class="va">amis_txf</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_alphamis_txf.html">get_alphamis_txf</a></span><span class="op">(</span>build<span class="op">=</span><span class="st">"hg38"</span><span class="op">)</span></span>
<span><span class="va">amis_txf</span></span></code></pre></div>
<pre><code><span><span class="co">## class: TabixFile </span></span>
<span><span class="co">## path: /home/vincent/.cache/R/BiocFileC.../a86b4ba7d3c4_AlphaMissense_hg38.tsv.gz</span></span>
<span><span class="co">## index: /home/vincent/.cache/R/Bioc.../a86b74da37cb_AlphaMissense_hg38.tsv.gz.tbi</span></span>
<span><span class="co">## isOpen: FALSE </span></span>
<span><span class="co">## yieldSize: NA</span></span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/Rsamtools/man/RsamtoolsFile-class.html" class="external-link">yieldSize</a></span><span class="op">(</span><span class="va">amis_txf</span><span class="op">)</span> <span class="op">=</span> <span class="fl">10L</span></span>
<span><span class="va">df10</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span>text<span class="op">=</span><span class="fu"><a href="https://rdrr.io/pkg/Rsamtools/man/scanTabix.html" class="external-link">scanTabix</a></span><span class="op">(</span><span class="va">amis_txf</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, h<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">amnames</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">df10</span><span class="op">)</span> <span class="op">=</span> <span class="va">amnames</span></span>
<span><span class="va">df10</span></span></code></pre></div>
<pre><code><span><span class="co">##    CHROM   POS REF ALT genome uniprot_id     transcript_id protein_variant</span></span>
<span><span class="co">## 1   chr1 69094   G   T   hg38     Q8NH21 ENST00000335137.4             V2L</span></span>
<span><span class="co">## 2   chr1 69094   G   C   hg38     Q8NH21 ENST00000335137.4             V2L</span></span>
<span><span class="co">## 3   chr1 69094   G   A   hg38     Q8NH21 ENST00000335137.4             V2M</span></span>
<span><span class="co">## 4   chr1 69095   T   C   hg38     Q8NH21 ENST00000335137.4             V2A</span></span>
<span><span class="co">## 5   chr1 69095   T   A   hg38     Q8NH21 ENST00000335137.4             V2E</span></span>
<span><span class="co">## 6   chr1 69095   T   G   hg38     Q8NH21 ENST00000335137.4             V2G</span></span>
<span><span class="co">## 7   chr1 69097   A   G   hg38     Q8NH21 ENST00000335137.4             T3A</span></span>
<span><span class="co">## 8   chr1 69097   A   C   hg38     Q8NH21 ENST00000335137.4             T3P</span></span>
<span><span class="co">## 9   chr1 69097   A   T   hg38     Q8NH21 ENST00000335137.4             T3S</span></span>
<span><span class="co">## 10  chr1 69098   C   A   hg38     Q8NH21 ENST00000335137.4             T3N</span></span>
<span><span class="co">##    am_pathogenicity      am_class</span></span>
<span><span class="co">## 1            0.2937 likely_benign</span></span>
<span><span class="co">## 2            0.2937 likely_benign</span></span>
<span><span class="co">## 3            0.3296 likely_benign</span></span>
<span><span class="co">## 4            0.2609 likely_benign</span></span>
<span><span class="co">## 5            0.2922 likely_benign</span></span>
<span><span class="co">## 6            0.2030 likely_benign</span></span>
<span><span class="co">## 7            0.0929 likely_benign</span></span>
<span><span class="co">## 8            0.1264 likely_benign</span></span>
<span><span class="co">## 9            0.0979 likely_benign</span></span>
<span><span class="co">## 10           0.1121 likely_benign</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="checking-against-gwas-hits-asthma-example">Checking against GWAS hits: asthma example<a class="anchor" aria-label="anchor" href="#checking-against-gwas-hits-asthma-example"></a>
</h2>
<p>We used the EBI GWAS catalog and searched for “sthma” in the <code>MAPPED_TRAIT</code> field. The resulting records are in a GRanges instance, accessible via <code>data(amentgr)</code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">amentgr</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">amentgr</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 3488</span></span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">amentgr</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"STRONGEST.SNP.RISK.ALLELE"</span>, <span class="st">"PUBMEDID"</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 3 ranges and 2 metadata columns:</span></span>
<span><span class="co">##       seqnames    ranges strand | STRONGEST.SNP.RISK.ALLELE  PUBMEDID</span></span>
<span><span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |               &lt;character&gt; &lt;numeric&gt;</span></span>
<span><span class="co">##   [1]       17  39909987      * |               rs2290400-C  27439200</span></span>
<span><span class="co">##   [2]        2  15780199      * |               rs2544523-T  24993907</span></span>
<span><span class="co">##   [3]        5 116261351      * |              rs10043228-T  24993907</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 23 sequences from GRCh38 genome; no seqlengths</span></span></code></pre>
<p>The coincidence of these GWAS hits with AlphaMissense results (<code>amis_txf</code> created above) can be computed quickly,.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Rsamtools" class="external-link">Rsamtools</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomeInfoDb" class="external-link">GenomeInfoDb</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">amentgr</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"chr"</span>, <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/seqinfo.html" class="external-link">seqlevels</a></span><span class="op">(</span><span class="va">amentgr</span><span class="op">)</span><span class="op">)</span> <span class="co"># if off line</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/Rsamtools/man/RsamtoolsFile-class.html" class="external-link">yieldSize</a></span><span class="op">(</span><span class="va">amis_txf</span><span class="op">)</span> <span class="op">=</span> <span class="cn">NA_integer_</span></span>
<span><span class="va">lk</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Rsamtools/man/scanTabix.html" class="external-link">scanTabix</a></span><span class="op">(</span><span class="va">amis_txf</span>, param<span class="op">=</span><span class="va">amentgr</span><span class="op">)</span></span></code></pre></div>
<p>Some of the positions with GWAS hits don’t correspond to results from AlphaMissense. These positions are empty components of the list returned by scanTabx. They are removed and the scanTabix results are converted to a data.frame.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ok</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">vapply</a></span><span class="op">(</span><span class="va">lk</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">&gt;</span><span class="fl">0</span>, <span class="fu"><a href="https://rdrr.io/r/base/logical.html" class="external-link">logical</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">cc</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"character"</span>, <span class="st">"character"</span>, <span class="st">"character"</span>, <span class="st">"character"</span>, <span class="st">"character"</span>, </span>
<span><span class="st">"character"</span>, <span class="st">"character"</span>, <span class="st">"character"</span>, <span class="st">"numeric"</span>, <span class="st">"character"</span><span class="op">)</span></span>
<span><span class="va">intsect</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">lk</span><span class="op">[</span><span class="va">ok</span><span class="op">]</span>, </span>
<span>   <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.delim</a></span><span class="op">(</span>text<span class="op">=</span><span class="va">x</span>, h<span class="op">=</span><span class="cn">FALSE</span>, colClasses<span class="op">=</span><span class="va">cc</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">intsect_df</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="va">intsect</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">amnames</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">intsect_df</span><span class="op">)</span> <span class="op">=</span> <span class="va">amnames</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">intsect_df</span>,<span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                          CHROM      POS REF ALT genome uniprot_id</span></span>
<span><span class="co">## chr4:38798089-38798089.1  chr4 38798089   T   A   hg38     Q15399</span></span>
<span><span class="co">## chr4:38798089-38798089.2  chr4 38798089   T   C   hg38     Q15399</span></span>
<span><span class="co">## chr4:38798089-38798089.3  chr4 38798089   T   G   hg38     Q15399</span></span>
<span><span class="co">##                              transcript_id protein_variant am_pathogenicity</span></span>
<span><span class="co">## chr4:38798089-38798089.1 ENST00000502213.6           N248I           0.1326</span></span>
<span><span class="co">## chr4:38798089-38798089.2 ENST00000502213.6           N248S           0.0737</span></span>
<span><span class="co">## chr4:38798089-38798089.3 ENST00000502213.6           N248T           0.1079</span></span>
<span><span class="co">##                               am_class</span></span>
<span><span class="co">## chr4:38798089-38798089.1 likely_benign</span></span>
<span><span class="co">## chr4:38798089-38798089.2 likely_benign</span></span>
<span><span class="co">## chr4:38798089-38798089.3 likely_benign</span></span></code></pre>
<p>At this point, <code>intsect_df</code> is the collection of <em>all</em> substitution scores at asthma GWAS hits. We focus on the substitutions reported as risk alleles.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">amentgr</span><span class="op">$</span><span class="va">RISK_ALLELE</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(.*-)"</span>, <span class="st">""</span>, <span class="va">amentgr</span><span class="op">$</span><span class="va">STRONGEST.SNP.RISK.ALLELE</span><span class="op">)</span></span></code></pre></div>
<p>To join the missense classes with the coincident GWAS hits we build a “key” for each table and then join.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">intsect_df</span><span class="op">$</span><span class="va">key</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">intsect_df</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">CHROM</span>, <span class="va">POS</span>, <span class="va">ALT</span>, sep<span class="op">=</span><span class="st">":"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">ament_df</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">amentgr</span><span class="op">)</span></span>
<span><span class="va">ament_df</span><span class="op">$</span><span class="va">key</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/with.html" class="external-link">with</a></span><span class="op">(</span><span class="va">ament_df</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">seqnames</span>, <span class="va">CHR_POS</span>, <span class="va">RISK_ALLELE</span>, sep<span class="op">=</span><span class="st">":"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">ia</span> <span class="op">=</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">inner_join</a></span><span class="op">(</span><span class="va">intsect_df</span>, <span class="va">ament_df</span>, by<span class="op">=</span><span class="st">"key"</span>, relationship <span class="op">=</span> <span class="st">"many-to-many"</span><span class="op">)</span></span>
<span><span class="va">iau</span> <span class="op">=</span> <span class="va">ia</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/duplicated.html" class="external-link">duplicated</a></span><span class="op">(</span><span class="va">ia</span><span class="op">$</span><span class="va">key</span><span class="op">)</span>,<span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">iau</span><span class="op">$</span><span class="va">am_class</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">##         ambiguous     likely_benign likely_pathogenic </span></span>
<span><span class="co">##                 1                15                 3</span></span></code></pre>
<p>Some of the information about the likely pathogenic substitutions that have been identified as hits for asthma is collected here:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">iau</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">am_class</span> <span class="op">==</span> <span class="st">"likely_pathogenic"</span><span class="op">)</span> <span class="op">|&gt;</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">CHR_ID</span>, <span class="va">CHR_POS</span>, <span class="va">STRONGEST.SNP.RISK.ALLELE</span>, </span>
<span>  <span class="va">OR.or.BETA</span>, <span class="va">MAPPED_TRAIT</span>, <span class="va">MAPPED_GENE</span>, <span class="va">am_class</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   CHR_ID   CHR_POS STRONGEST.SNP.RISK.ALLELE OR.or.BETA MAPPED_TRAIT</span></span>
<span><span class="co">## 1      5  14610200              rs16903574-G  1.0862118       asthma</span></span>
<span><span class="co">## 2      1  12115601               rs2230624-A  0.1618169       asthma</span></span>
<span><span class="co">## 3      9 128721272              rs11539209-A         NA       asthma</span></span>
<span><span class="co">##   MAPPED_GENE          am_class</span></span>
<span><span class="co">## 1     OTULINL likely_pathogenic</span></span>
<span><span class="co">## 2     TNFRSF8 likely_pathogenic</span></span>
<span><span class="co">## 3     ZDHHC12 likely_pathogenic</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Vincent Carey.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
